lib/article_status_manager.rb

```diff
@@ -12,25 +12,20 @@ def initialize(wiki = nil)
   # Entry points #
   ################
 
-  # Queries deleted state and namespace for all articles from current courses.
-  def self.update_article_status
-    threads = Course.current
-                    .in_groups(Replica::CONCURRENCY_LIMIT, false)
-                    .map.with_index do |course_batch, i|
-      Thread.new(i) do
-        course_batch.each do |course|
-          update_article_status_for_course(course)
-        end
-      end
-    end
-    threads.each(&:join)
-  end
-
   def self.update_article_status_for_course(course)
     Wiki.all.each do |wiki|
       next unless course.pages_edited.exists?(wiki_id: wiki.id)
-      course.pages_edited.where(wiki_id: wiki.id).find_in_batches do |article_batch|
-        new(wiki).update_status(article_batch)
+      # Updating only those articles which are updated more than  1 day ago
+      course.pages_edited
+            .where(wiki_id: wiki.id)
+            .where('articles.updated_at < ? OR articles.created_at = articles.updated_at',
```

Comment:
I don't think the `created_at` part is necessary here. Unlike a `Category`, where it gets created in a state of incomplete info (ie, there's no list of articles for it at the time it gets created), an `Article` record gets created in an up-to-date state at the time the first revisions to that article get imported. So if it was created less than a day ago (and updated only at that time) then it still doesn't need to be updated during this process.

Reply:
Done!